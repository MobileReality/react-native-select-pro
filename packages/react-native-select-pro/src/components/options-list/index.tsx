import React, { ComponentProps, useCallback } from 'react';
import {
    AccessibilityInfo,
    findNodeHandle,
    FlatList,
    StyleSheet,
    TouchableWithoutFeedback,
    View,
    ViewStyle,
} from 'react-native';
import { Portal } from '@gorhom/portal';

import { Portals } from '../../constants/portals';
import {
    BORDER_WIDTH,
    COLORS,
    ITEM_HEIGHT,
    MAX_HEIGHT_LIST,
    SHAPE,
} from '../../constants/styles';
import type { OptionalToRequired } from '../../helpers/types/OptionalToRequired';
import type { Position, State } from '../../state/types';
import type {
    OnOutsidePress,
    OnPressOptionType,
    OptionType,
} from '../../types';
import { NoOptions } from '../no-options';
import { Option } from '../option';
import { OptionsListWrapper } from '../options-list-wrapper';
import type { Select } from '../select';

type FromSelectComponentProps = Pick<
    ComponentProps<typeof Select>,
    | 'flatListProps'
    | 'optionSelectedStyle'
    | 'optionStyle'
    | 'optionTextStyle'
    | 'scrollToSelectedOption'
    | 'noOptionsText'
    | 'onSelect'
    | 'animated'
    | 'animationDuration'
    | 'optionsListStyle'
    | 'NoOptionsComponent'
    | 'OptionComponent'
    | 'searchable'
    | 'multiSelection'
>;

type OptionsListProps = OptionalToRequired<
    FromSelectComponentProps &
        Pick<
            State,
            | 'isOpened'
            | 'openedPosition'
            | 'optionsData'
            | 'selectedOption'
            | 'searchedOptions'
            | 'searchValue'
            | 'selectedOptionIndex'
        > & {
            onOutsidePress: OnOutsidePress;
            onPressOption: OnPressOptionType;
        } & Pick<Position, 'aboveSelectControl'>
>;

export const OptionsList = ({
    aboveSelectControl,
    flatListProps,
    onPressOption,
    selectedOption,
    animated,
    animationDuration,
    searchedOptions,
    searchValue,
    searchable,
    isOpened,
    onOutsidePress,
    multiSelection,
    openedPosition: { width, top, left },
    optionsData,
    optionSelectedStyle,
    optionStyle,
    optionTextStyle,
    noOptionsText,
    scrollToSelectedOption,
    onSelect,
    optionsListStyle,
    NoOptionsComponent,
    OptionComponent,
    selectedOptionIndex,
}: OptionsListProps) => {
    const selectedOptionTyped = selectedOption as OptionType;

    const flatList = useCallback(
        // TODO: remove any
        (node: any) => {
            const isScrollToSelectedOption =
                node &&
                scrollToSelectedOption &&
                selectedOptionIndex >= 0 &&
                typeof selectedOptionIndex === 'number';

            if (isScrollToSelectedOption) {
                node.scrollToIndex({
                    index: selectedOptionIndex,
                    animated: false,
                });
            }
        },
        [scrollToSelectedOption, selectedOptionIndex, onPressOption],
    );

    const measuredRef = useCallback(
        // TODO: remove any
        (node: any) => {
            if (node !== null) {
                const reactTag = findNodeHandle(node);
                if (reactTag) {
                    AccessibilityInfo.setAccessibilityFocus(reactTag);
                }
            }
        },
        [isOpened],
    );

    const resolveData = () => {
        if (!searchable) {
            return optionsData;
        }
        if (searchable && searchValue.length === 0) {
            return optionsData;
        }
        if (
            selectedOptionTyped &&
            searchValue?.length > 0 &&
            searchValue === selectedOptionTyped.label
        ) {
            return optionsData;
        }
        return searchedOptions;
    };

    const resolveIsSelected = (item: OptionType) => {
        if (!multiSelection) {
            return item.value === selectedOptionTyped?.value;
        }
        return (
            selectedOption &&
            (selectedOption as OptionType[]).find(
                (option) => item.value === option.value,
            )
        );
    };

    return (
        <>
            {isOpened && (
                <Portal hostName={Portals.SelectOutsideWrapper}>
                    <TouchableWithoutFeedback
                        accessibilityLabel="Close a dropdown from outside"
                        accessibilityRole="button"
                        onPress={onOutsidePress}
                    >
                        <View style={styles.modalOverlay} />
                    </TouchableWithoutFeedback>
                </Portal>
            )}
            <Portal hostName={Portals.Select}>
                <OptionsListWrapper
                    animated={animated}
                    animationDuration={animationDuration}
                    isOpened={isOpened}
                    wrapperStyles={[
                        styles.options,
                        optionsListStyle,
                        { top, left, width },
                        aboveSelectControl
                            ? styles.overflown
                            : styles.notOverflown,
                    ]}
                >
                    <FlatList
                        ref={flatList}
                        accessibilityLabel="Options list"
                        accessibilityState={{
                            expanded: isOpened,
                        }}
                        bounces={false}
                        data={resolveData()}
                        getItemLayout={(_data, index) => {
                            const height =
                                StyleSheet.flatten(optionStyle)?.height;
                            const isNumber = typeof height === 'number';
                            return {
                                length: isNumber ? height : ITEM_HEIGHT,
                                offset: isNumber
                                    ? height * index
                                    : ITEM_HEIGHT * index,
                                index,
                            };
                        }}
                        keyExtractor={({ value }) => value}
                        keyboardShouldPersistTaps="handled"
                        persistentScrollbar={true}
                        renderItem={({ item, index }) => {
                            const { value } = item;
                            const isSelected = !!resolveIsSelected(item);

                            return (
                                <Option
                                    key={value}
                                    ref={index === 0 ? measuredRef : undefined}
                                    OptionComponent={OptionComponent}
                                    isSelected={isSelected}
                                    option={item}
                                    optionSelectedStyle={optionSelectedStyle}
                                    optionStyle={optionStyle}
                                    optionTextStyle={optionTextStyle}
                                    optionIndex={index}
                                    onPressOption={onPressOption}
                                    onSelect={onSelect}
                                />
                            );
                        }}
                        {...flatListProps}
                        ListEmptyComponent={
                            NoOptionsComponent || (
                                <NoOptions noOptionsText={noOptionsText} />
                            )
                        }
                    />
                </OptionsListWrapper>
            </Portal>
        </>
    );
};

type Styles = {
    modalOverlay: ViewStyle;
    options: ViewStyle;
    notOverflown: ViewStyle;
    overflown: ViewStyle;
};

const styles = StyleSheet.create<Styles>({
    modalOverlay: {
        ...StyleSheet.absoluteFillObject,
        zIndex: 1,
    },
    options: {
        flex: 1,
        position: 'absolute',
        zIndex: 1,
        backgroundColor: COLORS.WHITE,
        borderWidth: BORDER_WIDTH,
        maxHeight: MAX_HEIGHT_LIST,
        elevation: 5,
    },
    notOverflown: {
        borderTopWidth: 0,
        borderBottomRightRadius: SHAPE,
        borderBottomLeftRadius: SHAPE,
    },
    overflown: {
        borderBottomWidth: 0,
        borderTopRightRadius: SHAPE,
        borderTopLeftRadius: SHAPE,
    },
});
