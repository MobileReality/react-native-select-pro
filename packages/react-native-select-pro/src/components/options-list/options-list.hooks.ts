import { useCallback } from 'react';
import type { View } from 'react-native';
import { AccessibilityInfo, findNodeHandle, StyleSheet } from 'react-native';

import { ITEM_HEIGHT } from '../../constants';
import { useOptionsListContext } from '../../context';
import { selectedOptionResolver } from '../../helpers';
import type { ItemLayout, OptionType } from '../../types';
import { isFlatOptionsType, isSectionOptionsType } from '../../types';

export const useOptionsList = <T>() => {
    const {
        optionsData,
        searchValue,
        searchedOptions,
        styles,
        isOpened,
        selectedOptionIndex,
        scrollToSelectedOption,
        flatListProps,
        selectedOption,
        sectionListProps,
        disabled,
        onPressOption,
        optionButtonProps,
        optionTextProps,
        pressableSelectedOption,
    } = useOptionsListContext();

    const { selectedOptionValue, selectedOptionLabel, selectedOptions } =
        selectedOptionResolver(selectedOption);

    const measuredRef = useCallback((node: View | null) => {
        if (node !== null) {
            const reactTag = findNodeHandle(node);
            if (reactTag) {
                AccessibilityInfo.setAccessibilityFocus(reactTag);
            }
        }
    }, []);

    const resolveData = useCallback(() => {
        if (!searchValue || (searchValue.length > 0 && searchValue === selectedOptionLabel)) {
            return optionsData;
        }
        return searchedOptions;
    }, [optionsData, searchValue, searchedOptions, selectedOptionLabel]);

    const findSelectedOption = useCallback(
        (item: OptionType<T>) => {
            if (selectedOptionValue) {
                return item.value === selectedOptionValue;
            }
            if (selectedOptions) {
                return selectedOptions.some((option) => item.value === option.value);
            }
            return false;
        },
        [selectedOptionValue, selectedOptions],
    );

    const findSelectedOptionIndex = useCallback(
        (item: OptionType<T>) => {
            if (isFlatOptionsType(optionsData)) {
                return optionsData.findIndex((option) => option.value === item.value);
            }

            return -1;
        },
        [optionsData],
    );

    const getItemLayout = useCallback(
        <T>(_data: T, index: number): ItemLayout<T> => {
            const height = StyleSheet.flatten(styles?.option?.container)?.height;
            const isNumber = typeof height === 'number';
            return {
                length: isNumber ? height : ITEM_HEIGHT,
                offset: isNumber ? height * index : ITEM_HEIGHT * index,
                index,
            };
        },
        [styles?.option?.container],
    );

    const resolvedData = resolveData();

    const isSectionedOptions = isSectionOptionsType(resolvedData);

    const initialScrollIndex =
        typeof selectedOptionIndex === 'number' && scrollToSelectedOption
            ? selectedOptionIndex
            : -1;

    const accessibilityState = {
        expanded: isOpened,
    };

    const { option: optionCustomStyles } = styles ?? {};

    const isDisabledResolveOption = useCallback(
        (isSelected: boolean) => {
            let isDisabledOption = false;

            if (disabled) {
                isDisabledOption = disabled;
            } else if (pressableSelectedOption) {
                isDisabledOption = false;
            } else if (isSelected) {
                isDisabledOption = true;
            }

            return isDisabledOption;
        },
        [disabled, pressableSelectedOption],
    );

    return {
        getItemLayout,
        measuredRef,
        resolvedData,
        findSelectedOption,
        findSelectedOptionIndex,
        scrollToSelectedOption,
        flatListProps,
        sectionListProps,
        selectedOption,
        optionCustomStyles,
        isSectionedOptions,
        initialScrollIndex,
        accessibilityState,
        disabled,
        onPressOption,
        optionButtonProps,
        optionTextProps,
        isDisabledResolveOption,
    };
};
