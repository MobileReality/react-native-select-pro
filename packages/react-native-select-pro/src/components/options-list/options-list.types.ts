import type { ComponentProps } from 'react';

import type { OptionalToRequired } from '../../helpers';
import type { Position, State } from '../../state/types';
import type { OnOutsidePress, OnPressOptionType } from '../../types';
import type { SelectStyles } from '../../types/styles';
import type { Select } from '../select';

type FromSelectComponentProps = Pick<
    ComponentProps<typeof Select>,
    | 'flatListProps'
    | 'scrollToSelectedOption'
    | 'noOptionsText'
    | 'onSelect'
    | 'animation'
    | 'NoOptionsComponent'
    | 'OptionComponent'
    | 'searchable'
    | 'multiSelection'
    | 'sectionListProps'
> &
    Pick<SelectStyles, 'optionsListStyles'>;

export type OptionsListProps = OptionalToRequired<
    FromSelectComponentProps &
        Pick<
            State,
            | 'isOpened'
            | 'openedPosition'
            | 'optionsData'
            | 'selectedOption'
            | 'searchedOptions'
            | 'searchValue'
            | 'selectedOptionIndex'
        > & {
            onOutsidePress: OnOutsidePress;
            onPressOption: OnPressOptionType;
        } & Pick<Position, 'aboveSelectControl'>
>;
