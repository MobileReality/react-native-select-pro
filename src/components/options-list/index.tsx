import React, { ComponentProps, useRef } from 'react';
import { FlatList, StyleSheet, TouchableWithoutFeedback, View, ViewStyle } from 'react-native';
import { Portal } from '@gorhom/portal';

import { Portals } from '../../constants/portals';
import { BORDER_WIDTH, COLORS, ITEM_HEIGHT, MAX_HEIGHT_LIST, SHAPE } from '../../constants/styles';
import type { OptionalToRequired } from '../../helpers';
import type { Position, State } from '../../state/types';
import type { OnOutsidePress, OnPressOptionType } from '../../types';
import { NoOptions } from '../no-options';
import { Option } from '../option';
import { OptionsListWrapper } from '../options-list-wrapper';
import type { Select } from '../select';

type FromSelectComponentProps = Pick<
    ComponentProps<typeof Select>,
    | 'flatListProps'
    | 'optionSelectedStyle'
    | 'optionStyle'
    | 'optionTextStyle'
    | 'scrollToSelectedOption'
    | 'noOptionsText'
    | 'onSelect'
    | 'animated'
    | 'animationDuration'
    | 'optionsListStyle'
    | 'NoOptionsComponent'
    | 'OptionComponent'
    | 'searchable'
>;

type OptionsListProps = OptionalToRequired<
    FromSelectComponentProps &
        Pick<
            State,
            | 'isOpened'
            | 'openedPosition'
            | 'optionsData'
            | 'selectedOption'
            | 'searchedOptions'
            | 'searchValue'
        > & {
            onOutsidePress: OnOutsidePress;
            onPressOption: OnPressOptionType;
        } & Pick<Position, 'aboveSelectControl'>
>;

export const OptionsList = ({
    aboveSelectControl,
    flatListProps,
    onPressOption,
    selectedOption,
    animated,
    animationDuration,
    searchedOptions,
    searchValue,
    searchable,
    isOpened,
    onOutsidePress,
    openedPosition: { width, top, left },
    optionsData,
    optionSelectedStyle,
    optionStyle,
    optionTextStyle,
    noOptionsText,
    scrollToSelectedOption,
    onSelect,
    optionsListStyle,
    NoOptionsComponent,
    OptionComponent,
}: OptionsListProps) => {
    const ref = useRef<FlatList>(null);

    const resolveData = () => {
        if (!searchable) {
            return optionsData;
        }
        if (searchable && searchValue.length === 0) {
            return optionsData;
        }
        if (selectedOption && searchValue?.length > 0 && searchValue === selectedOption.label) {
            return optionsData;
        }
        return searchedOptions;
    };

    return (
        <>
            {isOpened && (
                <Portal hostName={Portals.SelectOutsideWrapper}>
                    <TouchableWithoutFeedback
                        accessibilityLabel={'Close a dropdown from outside'}
                        onPress={onOutsidePress}>
                        <View style={styles.modalOverlay} />
                    </TouchableWithoutFeedback>
                </Portal>
            )}
            <Portal hostName={Portals.Select}>
                <OptionsListWrapper
                    animated={animated}
                    animationDuration={animationDuration}
                    isOpened={isOpened}
                    wrapperStyles={[
                        styles.options,
                        optionsListStyle,
                        { top, left, width },
                        aboveSelectControl ? styles.overflown : styles.notOverflown,
                    ]}>
                    <FlatList
                        accessibilityLabel={isOpened ? 'Options list' : ''}
                        bounces={false}
                        data={resolveData()}
                        getItemLayout={(_data, index) => {
                            const height = StyleSheet.flatten(optionStyle)?.height;
                            const isNumber = typeof height === 'number';
                            return {
                                length: isNumber ? height : ITEM_HEIGHT,
                                offset: isNumber ? height * index : ITEM_HEIGHT * index,
                                index,
                            };
                        }}
                        keyExtractor={({ value }) => value}
                        keyboardShouldPersistTaps="handled"
                        persistentScrollbar={true}
                        ref={ref}
                        renderItem={({ item, index }) => {
                            const { value } = item;
                            const isSelected = value === selectedOption?.value;
                            const isScrollToSelectedOption =
                                isSelected && ref.current && scrollToSelectedOption;

                            if (isScrollToSelectedOption) {
                                ref.current.scrollToIndex({
                                    index,
                                    animated: false,
                                });
                            }

                            return (
                                <Option
                                    OptionComponent={OptionComponent}
                                    isSelected={isSelected}
                                    key={value}
                                    onPressOption={onPressOption}
                                    onSelect={onSelect}
                                    option={item}
                                    optionSelectedStyle={optionSelectedStyle}
                                    optionStyle={optionStyle}
                                    optionTextStyle={optionTextStyle}
                                />
                            );
                        }}
                        {...flatListProps}
                        ListEmptyComponent={
                            NoOptionsComponent || <NoOptions noOptionsText={noOptionsText} />
                        }
                    />
                </OptionsListWrapper>
            </Portal>
        </>
    );
};

type Styles = {
    modalOverlay: ViewStyle;
    options: ViewStyle;
    notOverflown: ViewStyle;
    overflown: ViewStyle;
};

const styles = StyleSheet.create<Styles>({
    modalOverlay: {
        ...StyleSheet.absoluteFillObject,
        zIndex: 1,
    },
    options: {
        flex: 1,
        position: 'absolute',
        zIndex: 1,
        backgroundColor: COLORS.WHITE,
        borderWidth: BORDER_WIDTH,
        maxHeight: MAX_HEIGHT_LIST,
        elevation: 5,
    },
    notOverflown: {
        borderTopWidth: 0,
        borderBottomRightRadius: SHAPE,
        borderBottomLeftRadius: SHAPE,
    },
    overflown: {
        borderBottomWidth: 0,
        borderTopRightRadius: SHAPE,
        borderTopLeftRadius: SHAPE,
    },
});
