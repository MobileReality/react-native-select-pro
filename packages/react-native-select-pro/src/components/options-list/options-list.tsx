import React, { forwardRef, useCallback } from 'react';
import type { ListRenderItem, SectionListRenderItem, View } from 'react-native';

import { getReducedSectionData } from '../../helpers';
import type { OptionType, SectionOptionType } from '../../types';
import { isSectionOptionsType } from '../../types';
import { FlatOptionsList } from '../flat-options-list';
import { Option } from '../option';
import { OptionsListWrapper } from '../options-list-wrapper';
import { SectionOptionsList } from '../section-options-list';

import { useOptionsList } from './options-list.hooks';

export const OptionsList = forwardRef<View>((_, optionsListRef) => {
    const {
        getItemLayout,
        measuredRef,
        findSelectedOption,
        findSelectedOptionIndex,
        resolvedData,
        scrollToSelectedOption,
        sectionListProps,
        flatListProps,
        selectedOption,
        optionCustomStyles,
        initialScrollIndex,
        accessibilityState,
        disabled,
        onPressOption,
        optionButtonProps,
        optionTextProps,
        isDisabledResolveOption,
    } = useOptionsList();

    const isSectionedOptions = isSectionOptionsType(resolvedData);

    const renderSection: SectionListRenderItem<OptionType> = useCallback(
        ({ item, index, section }) => {
            const data = resolvedData as SectionOptionType[];
            const { value } = item;
            const isSelected = findSelectedOption(item);
            const sectionTitle = section?.title;
            const optionIndex = getReducedSectionData(data).indexOf(item);
            const sectionObject = {
                title: sectionTitle,
                index: data.findIndex((el) => el.title === sectionTitle),
            };
            const isDisabledOption = isDisabledResolveOption(isSelected);
            const sectionItem = { ...item, section: sectionObject };

            return (
                <Option
                    key={value}
                    ref={index === 0 ? measuredRef : undefined}
                    option={sectionItem}
                    isSelected={isSelected}
                    optionIndex={optionIndex}
                    overrideWithDisabledStyle={!!disabled}
                    optionButtonProps={optionButtonProps}
                    optionTextProps={optionTextProps}
                    optionCustomStyles={optionCustomStyles}
                    isDisabled={isDisabledOption}
                    onPressOption={onPressOption}
                />
            );
        },
        [
            disabled,
            findSelectedOption,
            isDisabledResolveOption,
            measuredRef,
            onPressOption,
            optionButtonProps,
            optionCustomStyles,
            optionTextProps,
            resolvedData,
        ],
    );

    const renderFlatItem: ListRenderItem<OptionType> = useCallback(
        ({ item, index }) => {
            const { value } = item;
            const isSelected = findSelectedOption(item);
            const optionIndex = findSelectedOptionIndex(item) ?? index;
            const isDisabledOption = isDisabledResolveOption(isSelected);

            return (
                <Option
                    key={value}
                    ref={index === 0 ? measuredRef : undefined}
                    option={item}
                    isSelected={isSelected}
                    optionIndex={optionIndex}
                    overrideWithDisabledStyle={!!disabled}
                    optionButtonProps={optionButtonProps}
                    optionTextProps={optionTextProps}
                    optionCustomStyles={optionCustomStyles}
                    isDisabled={isDisabledOption}
                    onPressOption={onPressOption}
                />
            );
        },
        [
            disabled,
            findSelectedOption,
            findSelectedOptionIndex,
            isDisabledResolveOption,
            measuredRef,
            onPressOption,
            optionButtonProps,
            optionCustomStyles,
            optionTextProps,
        ],
    );

    return (
        <OptionsListWrapper ref={optionsListRef}>
            {isSectionedOptions ? (
                <SectionOptionsList
                    resolvedData={resolvedData}
                    getItemLayout={getItemLayout}
                    renderItem={renderSection}
                    accessibilityState={accessibilityState}
                    selectedOption={selectedOption}
                    scrollToSelectedOption={scrollToSelectedOption}
                    sectionListProps={sectionListProps}
                    disabled={disabled}
                />
            ) : (
                <FlatOptionsList
                    initialScrollIndex={initialScrollIndex}
                    getItemLayout={getItemLayout}
                    renderItem={renderFlatItem}
                    accessibilityState={accessibilityState}
                    resolvedData={resolvedData}
                    flatListProps={flatListProps}
                    disabled={disabled}
                />
            )}
        </OptionsListWrapper>
    );
});

OptionsList.displayName = 'OptionsList';
