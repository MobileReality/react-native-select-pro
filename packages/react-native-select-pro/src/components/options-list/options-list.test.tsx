import React from 'react';

import { OptionsList } from './options-list';

describe('OptionsList', () => {
    it('should generate OptionsList snapshot', () => {
        const data = [{ label: 'test', value: 'test' }];
        const optionsList = (
            <OptionsList
                NoOptionsComponent={undefined}
                OptionComponent={undefined}
                aboveSelectControl={false}
                animation={false}
                flatListProps={{}}
                isOpened={false}
                multiSelection={false}
                noOptionsText={undefined}
                openedPosition={{ width: 0, top: 0, left: 0, aboveSelectControl: false }}
                optionsData={data}
                scrollToSelectedOption={false}
                searchValue=""
                searchable={false}
                searchedOptions={[]}
                selectedOption={data[0]}
                selectedOptionIndex={0}
                sectionListProps={{}}
                optionsListStyles={{}}
                onOutsidePress={() => {}}
                onPressOption={() => {}}
                onSelect={() => {}}
            />
        );
        expect(optionsList).toMatchSnapshot();
    });
});
