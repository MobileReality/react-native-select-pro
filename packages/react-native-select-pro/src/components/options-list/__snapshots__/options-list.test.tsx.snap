// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsList should generate OptionsList snapshot 1`] = `
<OptionsList
  aboveSelectControl={false}
  animation={false}
  flatListProps={{}}
  isOpened={false}
  multiSelection={false}
  onOutsidePress={[Function]}
  onPressOption={[Function]}
  onSelect={[Function]}
  openedPosition={
    {
      "aboveSelectControl": false,
      "left": 0,
      "top": 0,
      "width": 0,
    }
  }
  optionsData={
    [
      {
        "label": "test",
        "value": "test",
      },
    ]
  }
  optionsListStyles={{}}
  scrollToSelectedOption={false}
  searchValue=""
  searchable={false}
  searchedOptions={[]}
  sectionListProps={{}}
  selectedOption={
    {
      "label": "test",
      "value": "test",
    }
  }
  selectedOptionIndex={0}
/>
`;
